app/view_utils/money_view_utils.rb

```diff
@@ -0,0 +1,15 @@
+module MoneyViewUtils
+  extend ActionView::Helpers::NumberHelper
+
+  module_function
+
+  # Converts instance of Money to properly formatted and localized string for view purposes
+  # Replaces spaces with narrow non breaking spaces, we do not want monetary amounts to be split on separate lines
+  def to_humanized(m, locale)
```

Comment:
This comment might come a bit too late (since you have already replaced all humanize calls). The `locale` could be also optional and default to `I18n.locale`. That makes it a bit easier to the view code to call this function, since the caller doesn't need to pass the locale. The downside of course is that the function is then dependant of the global `I18n.locale` state.